posthog-node-ai/utils.ts

```diff
@@ -0,0 +1,209 @@
+import { PostHog } from "../posthog-node";
+import type { ChatCompletionCreateParamsBase } from 'openai/resources/chat/completions';
+
+export interface MonitoringParams {
+    posthog_distinct_id?: string;
+    posthog_trace_id?: string;
+    posthog_properties?: Record<string, any>;
+    posthog_privacy_mode?: boolean;
+    posthog_groups?: Record<string, any>;
+  }
+
+export const getModelParams = (params: ChatCompletionCreateParamsBase & MonitoringParams): Record<string, any> => {
+    const modelParams: Record<string, any> = {};
+    const paramKeys = [
+      'temperature',
+      'max_tokens',
+      'max_completion_tokens',
+      'top_p',
+      'frequency_penalty',
+      'presence_penalty',
+      'n',
+      'stop',
+      'stream',
+      'streaming',
+    ] as const;
+  
+    for (const key of paramKeys) {
+      if (key in params && (params as any)[key] !== undefined) {
+        modelParams[key] = (params as any)[key];
+      }
+    }
+    return modelParams;
+};
+  
+
+  export const extractCoreModelParams = (params: ChatCompletionCreateParamsBase & MonitoringParams, provider: string) => {
```

Comment:
Like in the Python SDK, I'm thinking this would be easier to maintain in the plugin server, with the same effect